src/backend/gog/library.ts

```diff
@@ -343,56 +347,76 @@ export class GOGLibrary {
       installPlatform
     ]
 
-    const res = await runGogdlCommand(commandParts, {
-      logMessagePrefix: 'Getting game metadata'
-    })
-    if (res.error) {
-      logError(['Failed to get game metadata for', `${appName}:`, res.error], {
+    const res = await runGogdlCommand(
+      commandParts,
+      createAbortController(appName),
+      {
+        logMessagePrefix: 'Getting game metadata'
+      }
+    )
+
+    deleteAbortController(appName)
+
+    if (res.abort) {
+      return
+    }
+
+    const errorMessage = (error: string) => {
+      logError(['Failed to get game metadata for', `${appName}:`, error], {
         prefix: LogPrefix.Gog
       })
     }
 
-    const gogInfo = JSON.parse(res.stdout)
-    const libraryArray = libraryStore.get('games', [{}]) as GameInfo[]
-    const gameObjectIndex = libraryArray.findIndex(
-      (value) => value.app_name === appName
-    )
+    if (res.error) {
+      errorMessage(res.error)
+    }
 
-    if (
-      !libraryArray[gameObjectIndex]?.gog_save_location &&
-      this.installedGames.get(appName) &&
-      this.installedGames.get(appName)?.platform !== 'linux'
-    ) {
-      gameData.gog_save_location = await this.getSaveSyncLocation(
-        appName,
-        this.installedGames.get(appName)!
+    try {
```

Comment:
I know I'm the only one with this opinion, but the solution to incomplete/incorrect type information that causes errors at runtime isn't "Just wrap everything in a `try {} catch`". I know it's not fun, but do research and actually fix the types. If we keep doing this, we might as well switch back to using JS

Reply:
One thing is not related to another.
Types will get most of the things but if a promise fails for unknown reasons, TS won't do anything.
TS won't run at runtime and won't get random issues.
For instance: An expected OS signal, a hang in the system, this type of thing. So having `try/catch `blocks will at least deal with some of these problems.
TS is for the dev not for the client.

Reply:
Let's keep this in the realm of "reasonably realistic" here. Having a random bit flip that erases a property from an object (I believe that's what you're implying by "hang in the system"), while also changing the extra bits of your ECC memory in the exact way so this doesn't get detected & corrected, is not realistic. And if it would be, who's to say it didn't just erase the try-catch block in the process? If everything's susceptible to this issue, why add protections at all then?
An "OS signal" (not sure what you mean there) that modifies your local JS variables does not exist. The OS can kill you, can pause you, and that's about it

What's far more realistic is:
- we're not taking care to check if a value satisfies all aspects of a type before using `as T`
- some of our types have properties that can be optional but aren't marked as such

Some parts of the code in this try-catch block can fail, yes. The system could run out of memory and GOGDL could fail to spawn, which would leave us with no `res.stdout`, and in turn `JSON.parse` throwing an error (on an unrelated note, if you *do* want to account for that here, there are a lot of cases elsewhere in the code where that's not being done)
But *at least* wrap just *that* in a try-catch then. When leaving it like this, it's impossible for someone who didn't run into the issue themselves to even guess what the problem was

Yes, TS doesn't "run" at runtime, because it doesn't "run" at all. It's a pre-processor, it prevents you from shooting yourself into the foot, but it still has ways (as described above) to disable this protection

Reply:
I agree that we need to be clear about what we are trying to catch with a try/catch block, wrapping too many lines in a try catch is confusing and if we know what can throw an exception we should only wrap that and add a comment about what was the error (then with that information we can try to prevent that in the future).

In languages with Exception and sub-classes for Exception this would mean the common best practice of don't catch `Exception`, catch the specific things that can be thrown. In the js world we don't have that but we could add comments about what we are trying to catch.